### PR TITLE
OSDOCS#9120: Adding Assisted Installer and Agent-based Installer entr…

### PR DESCRIPTION
**Pull request by mjpytlak**
Version(s):
4.15 +

Issue:
This issue addresses [osdocs-9120](https://issues.redhat.com/browse/OSDOCS-9120) and implements a refactored vSphere installation TOC, which is detailed in this Miro [board](https://miro.com/app/board/uXjVNbUhuKQ=/). This refactor has been approved by Stephanie Stout (CS) and Ju Lim (Senior Manager, Product Management).

Link to docs preview:

- [Installing a cluster on vSphere using the Assisted Installer](https://70279--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installing-vsphere-assisted-installer)
- [Installing a cluster on vSphere using the Agent-based installer](https://70279--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installing-vsphere-agent-based-installer)

QE review:
[N/A] QE has approved this change.

QE approval is not required for this change. While this PR adds two new assemblies to Installing on vSphere, the content of each assembly [1] [2] reuses existing approved docs. The new assemblies were added at the request of PM as part of the overall effort so that each type of installation method supported on vSphere was referenced from a single location.

Additional information:

[1] The abstract that is used for `installing/installing_vsphere/installing-vsphere-assisted-installer.adoc` is a combination of content that leveraged from the following existing content:

- `installing/installing_on_prem_assisted/installing-on-prem-assisted.adoc` 
- `modules/assisted-installer-using-the-assisted-installer.adoc`

[2] The abstract that is used for `installing/installing_vsphere/installing-vsphere-agent-based-installer.adoc` leverages existing documentation from `installing/installing_with_agent_based_installer/preparing-to-install-with-agent-based-installer.adoc`

### CODE REVIEW

File: installing/installing_vsphere/installing-vsphere-agent-based-installer.adoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+The Agent-based installation method provides the flexibility to boot your on-premises servers in any way that you choose. It combines the ease of use of the Assisted Installation service with the ability to run offline, including in air-gapped environments.
+
+Agent-based installation is a subcommand of the {product-title} installer. It generates a bootable ISO image containing all of the information required to deploy an {product-title} cluster with an available release image.
+
+[role="_additional-resources"]
+[id="additional-resources_installing-vsphere-agent-based-installer"]
+== Additional resources
+
+* xref:../../installing/installing_with_agent_based_installer/preparing-to-install-with-agent-based-installer.adoc#preparing-to-install-with-agent-based-installer[Preparing to install with the Agent-based installer]
```
Comment (skopacz1) [inline installing/installing_vsphere/installing-vsphere-agent-based-installer.adoc:17]: This looks good to me. If you want to be extra thorough you can add another link to the actual install procedure, "[Installing a cluster with Agent-based Installer](https://docs.openshift.com/container-platform/4.14/installing/installing_with_agent_based_installer/installing-with-agent-based-installer.html)", but I think just this one link to the prep page is enough as well.

Comment (skopacz1) [reply installing/installing_vsphere/installing-vsphere-agent-based-installer.adoc:17]: This also makes me realize I need to go through the Agent docs and capitalize "Installer"  😅

File: installing/installing_vsphere/installing-vsphere-assisted-installer.adoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+[id="installing-vsphere-assisted-installer"]
+= Installing a cluster on vSphere using the Assisted Installer
+include::_attributes/common-attributes.adoc[]
+:context: installing-vsphere-assisted-installer
+
+toc::[]
+
+You can install {product-title} on on-premise hardware or on-premise VMs by using the {ai-full}. Installing {product-title} by using the {ai-full} supports `x86_64`, `AArch64`, `ppc64le`, and `s390x` CPU architectures.
+
+The {ai-full} is a user-friendly installation solution offered on the Red Hat Hybrid Cloud Console. The {ai-full} supports the various deployment platforms with a focus on the following infrastructures:
```
Comment (skopacz1) [inline installing/installing_vsphere/installing-vsphere-assisted-installer.adoc:11]: Tiny nit that I just learned last week: apparently we should use a `{nbsp}` in between Red and Hat so the company name doesn't get broken up onto two separate lines, per the [RHSSG](https://redhat-documentation.github.io/supplementary-style-guide/#non-breaking-spaces)

```suggestion
The {ai-full} is a user-friendly installation solution offered on the Red{nbsp}Hat Hybrid Cloud Console. The {ai-full} supports the various deployment platforms with a focus on the following infrastructures:
```

**Review by skopacz1 — COMMENTED** [review]
Left a few comments but otherwise looks good to me!

File: installing/installing_vsphere/installing-vsphere-agent-based-installer.adoc (hunk 1 above)
Comment (mjpytlak) [reply installing/installing_vsphere/installing-vsphere-agent-based-installer.adoc:17]: I would prefer to leave this link only. I think that this topic is where the rubber meets the road in terms of introducing the installation method.

**Pull request merged**